### PR TITLE
feat(runtime-vapor): provide and inject

### PR DESCRIPTION
**Pull request by Ubugeeei**
related: #4 

Implemented provide/inject!
Almost all tests in runtime-core passed (except for the Options API).

Uncertainties:
- What to do with currentRenderingInstance
- Is it okay to pass the parent as an argument? ✅

### CODE REVIEW

**Comment by Ubugeeei** [comment]
Passing the parent instance as an argument or getting it within createComponentInstance using getCurrentInstance is a dilemma. The former seems better, but it might be a bit redundant.

**Comment by sxzz** [comment]
Why do we have to pass parent as an argument?  I think `currentInstance` is enough.

File: playground/src/props.js [hunk 1]
```diff
@@ -33,7 +34,7 @@ export default defineComponent({
       on(n0, 'click', () => handleClick)
       renderEffect(() => setText(n0, count.value))
       /** @type {any} */
-      const n1 = createComponent(child, [
+      const n1 = createComponent(child, getCurrentInstance(), [
```
Comment (sxzz) [inline playground/src/props.js:37]: I don't think it's a good idea, to pass parent instance as an argument. We should handle the logic inside of `createComponent`.

**Comment by Ubugeeei** [comment]
I've updated createComponentInstance to call getCurrentInstance!

The reason I thought passing arguments might be better is because it seemed somewhat implicit, and the original implementation (renderer.ts) handled it with the parentComponent argument, so I thought it might be better to align with that. I fundamentally don't grasp why it's like that...

**Review by sxzz — APPROVED** [review]

**Pull request merged**